utils/markdown.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package utils
+
+import (
+	"bytes"
+	"strings"
+
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/extension"
+	astExt "github.com/yuin/goldmark/extension/ast"
+	"github.com/yuin/goldmark/renderer"
+	"github.com/yuin/goldmark/util"
+)
+
+func StripMarkdown(markdown string) string {
+	md := goldmark.New(
+		goldmark.WithExtensions(extension.GFM),
+		goldmark.WithRenderer(
+			renderer.NewRenderer(renderer.WithNodeRenderers(
+				util.Prioritized(newNotificationRenderer(), 500),
+			)),
+		),
+	)
+
+	var buf bytes.Buffer
+	if err := md.Convert([]byte(markdown), &buf); err != nil {
+		return ""
+	}
+
+	return strings.TrimSpace(buf.String())
+}
+
+type notificationRenderer struct {
```

Comment:
Perhaps `markdownStripper` is a better name? [0/5](https://handbook.mattermost.com/company/about-mattermost/list-of-terms#0-5-1-5-2-5-3-5-4-5-5-5).

Reply:
I follow other renderer, they use postfix renderer and not all of markdown will be stripped in this function.